### ISSUE TITLE
java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Integer com.onesignal.OSUtils.getNetType()

### ISSUE THREAD

**Issue by gsksarath** [OneSignal/OneSignal-Android-SDK#800]
```
 E/AndroidRuntime: FATAL EXCEPTION: OS_REG_USER
    Process: com.example.app, PID: 19782
    java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Integer com.onesignal.OSUtils.getNetType()' on a null object reference
        at com.onesignal.OneSignal.registerUserTask(OneSignal.java:1286)
        at com.onesignal.OneSignal.access$1700(OneSignal.java:77)
        at com.onesignal.OneSignal$7.run(OneSignal.java:1254)
        at java.lang.Thread.run(Thread.java:764)

```

it can be easily reproducible by setting `OneSignal.provideUserConsent(false); ` then lock & unlock the phone (NOKIA 5.1 plus)

build.gradle project level
```
classpath 'com.android.tools.build:gradle:3.2.1'
classpath 'com.google.gms:google-services:4.1.0'
```
build.gradle app level
```

buildscript {
    repositories {
        maven { url 'https://plugins.gradle.org/m2/'}
    }
    dependencies {
        classpath 'gradle.plugin.com.onesignal:onesignal-gradle-plugin:[0.12.2, 0.99.99]'
    }
}
apply plugin: 'com.onesignal.androidsdk.onesignal-gradle-plugin'

apply plugin: 'com.android.application'

android {
    compileSdkVersion 28

    defaultConfig {
        applicationId "com.example.app"
        minSdkVersion 17
        targetSdkVersion 28
        manifestPlaceholders = [manifestApplicationId: "${applicationId}",
                            
                                onesignal_google_project_number: "REMOTE"]

        multiDexEnabled true
        resConfigs "en"
    }

    buildTypes {

        debug{
            debuggable true
            minifyEnabled false
            resValue("string", "PORT_NUMBER", "8081")
        }
        release {
            debuggable false
            minifyEnabled true
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-project.txt'
        }
    }
    dexOptions {
        javaMaxHeapSize "4g"
            jumboMode = true

    }


    repositories {
        google()
        jcenter()
        maven { url "https://s3.amazonaws.com/moat-sdk-builds" }
        maven { url 'https://maven.google.com' }
    
        maven { url 'https://jitpack.io' }

        mavenCentral()
    }

    useLibrary  'org.apache.http.legacy'

    compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }

}

dependencies {



    implementation fileTree(dir: 'libs', include: ['*.jar'])


    

    implementation 'com.android.support:appcompat-v7:28.0.0'
    implementation 'com.android.support:cardview-v7:28.0.0'
    implementation 'com.android.support:recyclerview-v7:28.0.0'
    implementation 'com.android.support:design:28.0.0'
    implementation 'com.android.support:preference-v7:28.0.0'
    implementation 'com.android.support:customtabs:28.0.0'
    implementation 'com.android.volley:volley:1.1.1'
    implementation 'com.onesignal:OneSignal:[3.10.9, 3.99.99]'
    implementation 'com.android.support:multidex:1.0.3'



    implementation 'com.google.firebase:firebase-messaging:19.0.1'
    implementation 'com.google.firebase:firebase-core:17.0.0'
    implementation 'com.google.firebase:firebase-auth:18.0.0'
    implementation 'com.google.firebase:firebase-database:18.0.0'


    implementation 'com.google.android.libraries.places:places:1.1.0'
    implementation 'com.google.android.gms:play-services-location:17.0.0'
    implementation 'com.google.android.gms:play-services-basement:17.0.0'
    implementation 'com.google.android.gms:play-services-auth:17.0.0'


    implementation 'com.google.android.ads.consent:consent-library:1.0.7'


}
apply plugin: 'com.google.gms.google-services'


```

**Comment by rgomezp** [comment]
Howdy!
All your firebase dependencies should be on the same version (e.g: 17.0.0). Could you try setting them all to the same version?

**Comment by gsksarath** [comment]
Getting same error even i set all firebase dependencies to 17.00.

**Comment by rgomezp** [comment]
It appears you aren't actually using the latest version. Can you make sure you're using 3.11.1 please? Cheers

**Comment by gsksarath** [comment]
Ok let me know what firebase SDK version should I have to use for onesignal
3.11.1

**Comment by jkasten2** [comment]
@gsksarath Thanks for the details, the crash stack trace doesn't point to a firebase issue. It's possible a change in version 3.11.0 or 3.11.1 of the OneSignal SDK could be the issue.

Can you try to reproduce the the specific crash in version 3.11.0 and 3.10.?

**Comment by gsksarath** [comment]
yes, i can easily reproduce in both 3.11.0 and 3.11.1 too.

**Comment by gsksarath** [comment]
@jkasten2 It's seems the issue #810 & #800 are same

**Comment by emmanuel-marty** [comment]
I reverted to SDK 3.10.9 as this is extremely widespread (crashing over 10% of user sessions!) and it seems to have appeared in 3.11.0 or 3.11.1.

**Comment by gsksarath** [comment]
@jkasten2 any update on this?

**Comment by T-T-P** [comment]
I too am observing this issue in my Xamarin app running v3.4.0 (latest stable) but only seeing the issue occur in the Android builds. I assume getNetType() is attempting to determine the connection type and that perhaps the OS is switching at that particular point or something? At present observing on Pixel devices and all Samsung Galaxy 6/7/8 and 9 devices on multiple versions of Android.

**Comment by rgomezp** [comment]
@gsksarath @T-T-P @emmanuel-marty ,
We will be releasing a fix for this very soon. Thanks for your patience

**Comment by jkasten2** [comment]
This issue is now resolved in the just released in 3.11.2 SDK
https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/3.11.2
